ocw/lib/gke.py

```diff
@@ -0,0 +1,77 @@
+import os
+import subprocess
+from datetime import datetime, timezone
+import kubernetes
+import googleapiclient.discovery
+from google.oauth2 import service_account
+from ocw.lib.gce import GCE
+
+
+class GKE(GCE):
+    __instances = {}
+
+    def __new__(cls, vault_namespace):
+        if vault_namespace not in GKE.__instances:
+            GKE.__instances[vault_namespace] = self = object.__new__(cls)
+            self.__gke_client = None
+            self.__kubectl_client = {}
+            self.init()
+        return GKE.__instances[vault_namespace]
+
+    def get_project_id(self):
+        return self.get_data()["project_id"]
+
+    def gke_client(self):
+        if self.__gke_client is None:
+            self.private_key_data = self.get_data()
+            credentials = service_account.Credentials.from_service_account_info(self.private_key_data)
+            self.__gke_client = googleapiclient.discovery.build('container', 'v1', credentials=credentials)
+
+        return self.__gke_client
+
+    def kubectl_client(self, zone, cluster):
+        cluster_name = cluster["name"]
+        kube_dir = "/root/.kube"
+        kubeconfig = f"{kube_dir}/gke_config_{zone}_{cluster_name}"
+        try:
+            self.cmd_exec(f"gcloud auth login  --project={self.get_project_id()} --cred-file=/var/pcw/qac/GKE.json --quiet")
```

Comment:
`var/pcw/qac/GKE.json` - "qac" is NO GO for sure as it is namespace which will be different for let's say `sapha`

also on top of that I would do more tight coupling with https://github.com/SUSE/pcw/blob/master/ocw/lib/provider.py#L20 where we basically defining from where we reading this . Maybe create method in provider ? e.g. `get_creds_location` ( feel free to give better name )

Reply:
Solved